test.js

```diff
@@ -179,6 +178,81 @@ describe('capture-exit', function() {
           expect(exitWasCalled).to.equal(1);
         });
       });
+
+      it('process.exit with an error while natural exit completes', function() {
+        var exitWasCalled = 0;
+        var onExitWasCalled = 0;
+        var exitCode;
+
+        process.exit = function stubExit(code) {
+          exitWasCalled++;
+          exitCode = code;
+        };
+
+        var deferred;
+        exit.captureExit();
+
+        exit.onExit(function() {
+          onExitWasCalled++;
+          deferred = RSVP.defer();
+          return deferred.promise;
+        });
+
+        process.exit();
+
+        return delay(10).then(function() {
+          expect(onExitWasCalled).to.equal(1);
+          process.exit('someErrorCode');
+
+          return delay(10).then(function() {
+            deferred.resolve();
+          });
+        }).then(() => {
+          return delay(0).then(function() {
```

Comment:
Honestly I'm not sure why do we need to make this delay. The test fails without it